src/librustdoc/html/render/mod.rs

```diff
@@ -2410,3 +2429,206 @@ fn collect_paths_for_type(first_ty: clean::Type, cache: &Cache) -> Vec<String> {
+        let (line_offset, _) = min_loc.enclosing_item.line_span;
+        let byte_ceiling =
+            call_data.locations.iter().map(|loc| loc.enclosing_item.byte_span.1).max().unwrap();
+
+        // The output code is limited to that byte range.
+        let contents_subset = &contents[(byte_offset as usize)..(byte_ceiling as usize)];
+
+        // The call locations need to be updated to reflect that the size of the program has changed.
+        // Specifically, the ranges are all subtracted by `byte_offset` since that's the new zero point.
+        let (byte_ranges, line_ranges): (Vec<_>, Vec<_>) = call_data
+            .locations
+            .iter()
+            .map(|loc| {
+                let (byte_lo, byte_hi) = loc.call_expr.byte_span;
+                let (line_lo, line_hi) = loc.call_expr.line_span;
+                (
+                    (byte_lo - byte_offset, byte_hi - byte_offset),
+                    (line_lo - line_offset, line_hi - line_offset),
+                )
+            })
+            .unzip();
+
+        let (init_min, init_max) = line_ranges[0];
+        let line_range = if init_min == init_max {
+            format!("line {}", init_min + line_offset + 1)
+        } else {
+            format!("lines {}-{}", init_min + line_offset + 1, init_max + line_offset + 1)
+        };
+
+        let edition = cx.shared.edition();
+        write!(
+            w,
+            r#"<div class="scraped-example" data-locs="{locations}" data-offset="{offset}">
+                <div class="scraped-example-title">
+                   {name} (<a href="{root}{url}" target="_blank">{line_range}</a>)
+                </div>
+                <div class="code-wrapper">"#,
+            root = cx.root_path(),
+            url = call_data.url,
+            name = call_data.display_name,
+            line_range = line_range,
+            offset = line_offset,
+            // The locations are encoded as a data attribute, so they can be read
+            // later by the JS for interactions.
+            locations = serde_json::to_string(&line_ranges).unwrap(),
+        );
+        write!(w, r#"<span class="prev">&pr;</span> <span class="next">&sc;</span>"#);
+        write!(w, r#"<span class="expand">&varr;</span>"#);
+
+        // Look for the example file in the source map if it exists, otherwise return a dummy span
+        let file_span = (|| {
+            let source_map = tcx.sess.source_map();
+            let crate_src = tcx.sess.local_crate_source_file.as_ref()?;
+            let abs_crate_src = crate_src.canonicalize().ok()?;
+            let crate_root = abs_crate_src.parent()?.parent()?;
+            let rel_path = path.strip_prefix(crate_root).ok()?;
+            let files = source_map.files();
+            let file = files.iter().find(|file| match &file.name {
+                FileName::Real(RealFileName::LocalPath(other_path)) => rel_path == other_path,
+                _ => false,
+            })?;
```

Comment:
This also looks sketchy and will break with --remap-path-prefix or if the same file is used twice with different `#[path]`s.

Reply:
Right now, the behavior is that if you do:

```rust
// foobar/src/a.rs
pub fn foo() {}

// foobar/src/lib.rs
#[path = "a.rs"]
pub mod b;
#[path = "a.rs"]
pub mod c;

// foobar/examples/ex.rs
fn main() {
  foobar::b::foo();
  foobar::c::foo();
}
```

Then the first call `b::foo()` will show up on the doc page for `b::foo()`, and the second call `c::foo()` will show up on the doc page for `c::foo()`. Is that different than what you would expect?

Reply:
Oh, perfect! That's not what I'd expect from the code but it sounds like the right behavior :) can you add that as a test case?